src/test/java/com/iexec/common/utils/FeignBuilderTest.java

```diff
@@ -37,6 +38,14 @@ void stopServer() {
         mockServer.stop();
     }
 
+    @Test
+    void testBuilderCreation() {
+        assertNotNull(FeignBuilder.createBuilder(Logger.Level.FULL)
+                .target(FeignTestClient.class, "http://localhost:" + mockServer.getPort()));
+        assertNotNull(FeignBuilder.createBuilder(Logger.Level.FULL, "username", "password")
+                .target(FeignTestClient.class, "http://localhost:" + mockServer.getPort()));
```

Comment:
Is it possible to check the username and password values in the returned instance?

Reply:
I don't think so.
https://javadoc.io/doc/io.github.openfeign/feign-core/latest/feign/Feign.Builder.html

We need to trust feign library implementation.
Tests to check authentication works as intended can only be done in integration tests.

Reply:
Everything will come into place with the next batch of pull requests